sql/src/main/java/org/apache/druid/sql/calcite/rel/DruidQuery.java

```diff
@@ -126,15 +151,17 @@ public DruidQuery(
       final boolean finalizeAggregations
   )
   {
-    this.dataSource = dataSource;
-    this.outputRowType = partialQuery.leafRel().getRowType();
-    this.sourceRowSignature = sourceRowSignature;
-    this.virtualColumnRegistry = VirtualColumnRegistry.create(sourceRowSignature);
-    this.plannerContext = plannerContext;
+    final RelDataType outputRowType = partialQuery.leafRel().getRowType();
+    final VirtualColumnRegistry virtualColumnRegistry = VirtualColumnRegistry.create(sourceRowSignature);
 
     // Now the fun begins.
+    final DimFilter filter;
+    final Projection selectProjection;
+    final Grouping grouping;
+    final Sorting sorting;
+
     if (partialQuery.getWhereFilter() != null) {
-      this.filter = Preconditions.checkNotNull(
+      filter = Preconditions.checkNotNull(
```

Comment:
Maybe add a description here since the pre-condition is on a computed value.

Similar comments in the rest of this file